apps/src/templates/teacherDashboard/EditSectionForm.jsx

```diff
@@ -177,7 +130,147 @@ export default connect(state => ({
   validAssignments: state.teacherSections.validAssignments,
   primaryAssignmentIds: state.teacherSections.primaryAssignmentIds,
   sections: state.teacherSections.sections,
-}))(EditSectionForm);
+  section: state.teacherSections.sectionBeingEdited,
+  isSaveInProgress: state.teacherSections.saveInProgress,
+}), {
+  editSectionProperties,
+  handleSave: finishEditingSection,
+  handleClose: cancelEditingSection,
+})(EditSectionForm);
+
+const FieldProps = {
+  value: PropTypes.any,
+  onChange: PropTypes.func.isRequired,
+  disabled: PropTypes.bool,
+};
+
+const SectionNameField = ({value, onChange, disabled}) => (
+  <div>
+    <FieldName>
+      {i18n.sectionName()}
+    </FieldName>
+    <FieldDescription>
+      {i18n.addSectionName()}
+    </FieldDescription>
+    <input
+      value={value}
+      onChange={event => onChange(event.target.value)}
+      style={style.sectionNameInput}
+      disabled={disabled}
+    />
+  </div>
+);
+SectionNameField.propTypes = FieldProps;
+
+const GradeField = ({value, onChange, validGrades, disabled}) => {
+  const gradeOptions = [""]
+    .concat(validGrades)
+    .map(grade => ({
+      value: grade,
+      text: grade === 'Other' ? 'Other/Mixed' : grade,
```

Comment:
Sure, question for @poorvasingal: priority of localizing the "Other/Mixed" option in grade selector?  And when a teacher picks this, should it also show up as "Other/Mixed" in the section table, or just as "Other"?